src/mongodb/scores.controller.ts

```diff
@@ -699,137 +1572,1444 @@ export class ScoresController {
+      contentLevel = 'L4';
+      complexityLevel = ["C1", "C2", "C3"]
+    }
 
-  @ApiExcludeEndpoint(true)
-  @Get(':id')
-  findOne(@Param('id') id: string) {
-    return this.scoresService.findOne(+id);
-  }
+    let getGetTargetCharArr = getGetTarget.filter((getGetTargetEle, index) => {
+      if (gettargetlimit > 0 && index >= gettargetlimit) {
+        return false;
+      }
+      return true;
+    }).map(charData => {
+      return charData.character
+    });
+
+    console.log(gettargetlimit);
+
+    const url = process.env.ALL_CONTENT_SERVICE_API;
+
+    const textData = {
+      "tokenArr": getGetTargetCharArr,
+      "language": language || "ta",
+      "contentType": "char",
+      "limit": contentlimit || 5,
+      "tags": tags,
+      "cLevel": contentLevel,
+      "complexityLevel": complexityLevel
+    };
+
+    const newContent = await lastValueFrom(
+      this.httpService.post(url, JSON.stringify(textData), {
+        headers: {
+          'Content-Type': 'application/json',
+        }
+      }).pipe(
+        map((resp) => resp.data)
+      )
+    );
+
+    let contentArr;
+    let contentForTokenArr;
+
+    if (newContent.data.hasOwnProperty('wordsArr')) {
+      contentArr = newContent.data.wordsArr;
+    } else {
+      contentArr = [];
+    }
 
-  @ApiExcludeEndpoint(true)
-  @Get('/byuser/:id')
-  findbyUser(@Param('id') id: string) {
-    return this.scoresService.findbyUser(id);
+    if (newContent.data.hasOwnProperty('contentForToken')) {
+      contentForTokenArr = newContent.data.contentForToken;
+    } else {
+      contentForTokenArr = [];
+    }
+
+    return { content: contentArr, contentForToken: contentForTokenArr, getTargetChar: getGetTargetCharArr, currentLevel: currentLevel, totalTargets: totalTargets };
   }
 
-  @ApiExcludeEndpoint(true)
-  @Get('/bysession/:id')
-  findbySession(@Param('id') id: string) {
-    return this.scoresService.findbySession(id);
+  @ApiParam({
+    name: "userId",
+    example: "2020076506"
+  })
+  @Get('GetContent/word/session/:sessionId')
+  @ApiOperation({ summary: 'Get a set of words for the session to practice, upon feeding the Get Target Chars to Content Algorithm by user id' })
+  @ApiResponse({
+    status: 200,
+    description: 'Success response with Get content and GetTarget chars for user id',
+    schema: {
+      properties: {
+        content: { type: 'string' },
+        getTargetChar: { type: 'string' }
+      }
+    },
+  })
+  async GetContentWordbySession(@Param('sessionId') id: string, @Query('language') language: string, @Query() { contentlimit = 5 }, @Query() { gettargetlimit = 5 }, @Query() { tags }) {
+
+    let getGetTarget = await this.scoresService.getTargetsBySession(id, language);
+    let validations = await this.scoresService.getAssessmentRecords(id);
+
+    let totalTargets = getGetTarget.length;
+    let totalValidation = validations.length;
+
+    let currentLevel = 'No Level Matched';
+
+    if (totalTargets >= 30) {
```

Comment:
create a function to manage currentLevel determination.  Also 30s and 15s should be configurable.